apps/web/components/eventtype/EventAvailabilityTab.tsx

```diff
@@ -278,25 +280,152 @@ const EventTypeSchedule = ({ eventType }: { eventType: EventTypeSetup }) => {
   );
 };
 
+const TeamMemberSchedule = ({ host, index }: { host: Host; index: number }) => {
+  const { t } = useLocale();
+
+  const formMethods = useFormContext<FormValues>();
+  const { setValue, getValues } = formMethods;
+
+  const { data, isPending } = trpc.viewer.availability.schedule.getAllSchedulesByUserId.useQuery({
+    userId: host.userId,
+  });
+
+  if (isPending) {
+    return <Spinner className="mt-3 h-6 w-6" />;
+  }
+
+  const schedules = data?.schedules;
+  const options = schedules?.map((schedule) => ({
+    value: schedule.id,
+    label: schedule.name,
+    isDefault: schedule.isDefault,
+    isManaged: false,
+  }));
+
+  //Set to defaultSchedule if Host Schedule is not previously selected
+  const scheduleId = getValues(`hosts.${index}.scheduleId`);
+  const availability = getValues(`hosts.${index}.availability`);
+  const value = options?.find((option) =>
+    scheduleId
+      ? option.value === scheduleId
+      : availability
+      ? option.value === availability?.value
+      : option.value === schedules?.find((schedule) => schedule.isDefault)?.id
+  );
+
+  if (!scheduleId) {
+    setValue(`hosts.${index}.scheduleId`, value?.value || null, { shouldDirty: true });
+    setValue(`hosts.${index}.availability`, value || null, { shouldDirty: true });
+  }
+
+  if (!availability) {
+    setValue(`hosts.${index}.availability`, value || null, { shouldDirty: false });
+  }
```

Comment:
You can also clean this up and use the value directly inside the Select.

Reply:
scheduleId has to be initialized, else 'null' will be sent to backend when -> new host is added in Assignment Tab but user does not change the default schedule selection. As OnChange() triggers only if there is change in selection.